core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/CarbonUtils.java

```diff
@@ -1255,6 +1255,17 @@ public static String getProxyContextPath(boolean isWorkerNode) {
 		return proxyContextPath;
 	}
 
+    /**
+     * Checks the incoming request transport
+     *
+     * @return true if the incoming request is from a local transport
+     */
+    public static boolean isRequestedFromLocalTransport() {
+        MessageContext msgCtx = MessageContext.getCurrentMessageContext();
```

Comment:
getThreadLocalCarbonContext() is used in CC. Here I am using MessageContext to review incoming request, hence using MessageContext.getCurrentMessageContext();